### PR TITLE
Check: Reference problems concerning structs

### PR DESCRIPTION
**Pull request by LianaHus**
PT story # 86149672

### CODE REVIEW

**Comment by LianaHus** [comment]
I need to rebase

**Comment by LianaHus** [comment]
rebased

File: test/libsolidity/SolidityEndToEndTest.cpp [hunk 1]
```diff
@@ -4023,6 +4023,92 @@ BOOST_AUTO_TEST_CASE(overwriting_inheritance)
...
+			testStruct data1;
+			testStruct data2;
+			testStruct data3;
+			function test()
+			{
+				data1.m_value = 2;
+			}
+			function assign() returns (uint ret_local, uint ret_global, uint ret_global3, uint ret_global1)
+			{
+				testStruct x = data1; //x is a reference data1..m_value == 2 as well as x.m_value = 2
```
Comment (chriseth) [inline test/libsolidity/SolidityEndToEndTest.cpp]: double `.`

File: test/libsolidity/SolidityEndToEndTest.cpp [hunk 2]
```diff
@@ -4023,6 +4023,92 @@ BOOST_AUTO_TEST_CASE(overwriting_inheritance)
...
+				data2 = data1; // should copy data. data2.m_value == 2
+
+				ret_local = x.m_value; // = 2
+				ret_global = data2.m_value; // = 2
+
+				x.m_value = 3;
+				data3 = x; //should copy the data. data3.m_value == 3
+				ret_global3 = data3.m_value; // = 3
+				ret_global1 = data1.m_value; // =3
```
Comment (chriseth) [inline test/libsolidity/SolidityEndToEndTest.cpp]: space and perhaps a comment: changed due to the assignment to x.m_value

File: test/libsolidity/SolidityEndToEndTest.cpp [hunk 3]
```diff
@@ -4023,6 +4023,92 @@ BOOST_AUTO_TEST_CASE(overwriting_inheritance)
...
+			function test()
+			{
+				data1.m_value = 2;
+			}
+			function deleteMember() returns (uint ret_value)
+			{
+				testStruct x = data1; //should not copy the data. data1.m_value == 2 but x.m_value = 0
+				x.m_value = 4;
+				delete x.m_value;
+				ret_value = x.m_value;
```
Comment (chriseth) [inline test/libsolidity/SolidityEndToEndTest.cpp]: Please use `data1.m_value` here to check that it really deleted in storage.

**Pull request merged**